source/core_sntp_client.c

```diff
@@ -312,39 +320,50 @@ static SntpStatus_t sendSntpPacket( const UdpTransportInterface_t * pNetworkIntf
         {
             LogError( ( "Unable to send request packet: Transport send failed. "
                         "ErrorCode=%ld.", ( long int ) bytesSent ) );
-            sendError = true;
+            status = SntpErrorNetworkFailure;
         }
-
-        /* Partial sends are not supported by UDP, which only supports sending the entire datagram as a whole.
-         * Thus, if the transport send function returns status representing partial send, it will be treated as failure. */
-        else if( ( bytesSent > 0 ) && ( bytesSent != ( int32_t ) packetSize ) )
-        {
-            LogError( ( "Unable to send request packet: Transport send returned unexpected bytes sent. "
-                        "ReturnCode=%ld, ExpectedCode=%u", ( long int ) bytesSent, packetSize ) );
-
-            sendError = true;
-        }
-        else
+        else if( bytesSent == 0 )
         {
             /* No bytes were sent over the network. Retry send if we have not timed out. */
+
             SntpTimestamp_t currentTime;
+            uint64_t elapsedTimeMs;
 
             getTimeFunc( &currentTime );
 
             /* Calculate time elapsed since last data was sent over network. */
-            timeSinceLastSendMs = calculateElapsedTimeMs( &currentTime, &lastSendTime );
+            elapsedTimeMs = calculateElapsedTimeMs( &currentTime, &lastSendTime );
 
             /* Check for timeout if we have been waiting to send any data over the network. */
-            if( timeSinceLastSendMs >= SNTP_SEND_RETRY_TIMEOUT_MS )
+            if( elapsedTimeMs >= timeoutMs )
```

Comment:
```suggestion
            if( elapsedTimeMs >= (uint64_t) timeoutMs )
```

Reply:
There is no risk of comparing between `uint64_t` and `uint32_t` as there is no loss of information (due to type promotion of the `uint32_t` operand). I verified that Coverity does not raise an issue.